### PR TITLE
Upgrade to Golang Version 1.22.6 and Golangci lint version v1.61.0

### LINKED ISSUE #3949: Upgrade to Golang 1.22.6
Steps to upgrade Golang version:
- [x] Update `go.mod` and `go.sum`. At the root of the directory, run:
    - [x] `find . -name 'go.mod' -not -path '*/\.*' -execdir go mod edit -go=<NEW_GOLANG_VERSION_WITHOUT_PATCH> \;`
    - [x] `find . -name 'go.mod' -not -path '*/\.*' -execdir go mod tidy \;`

- [x] Update the Dockerfiles for `build` directory. At the root of the directory, run: 
    
    `find build -type f \( -not -path '*/\.*' -and -not -path 'build/tmp/*' \) -exec sed -i 's/GO_VERSION=[0-9]\+\.[0-9]\+\.[0-9]\+/GO_VERSION=<NEW_GOLANG_VERSION>/g' {} \;`
    
- [x] Update the Dockerfiles for `examples` directory. At the root of the directory, run:     
    - [x] `find examples -name Dockerfile -exec sed -i 's/golang:[0-9]\+\.[0-9]\+-alpine/golang:<NEW_GOLANG_VERSION_WITHOUT_PATCH>-alpine/g' {} \;`
    - [x] `find examples \( -name Dockerfile -o -name Dockerfile.windows \) -exec sed -i 's/golang:[0-9]\+\.[0-9]\+\.[0-9]\+/golang:<NEW_GOLANG_VERSION>/g' {} \;`
    
- [x] Update the example images tag. At `build` directory, run:
    - [x] `make bump-image IMAGENAME=allocation-endpoint-proxy VERSION=<current-image-version>`
    - [x] `make bump-image IMAGENAME=autoscaler-webhook VERSION=<current-image-version>`
    - [x] `make bump-image IMAGENAME=crd-client VERSION=<current-image-version>`
    - [x] `make bump-image IMAGENAME=custom-controller VERSION=<current-image-version>`
    - [x] `make bump-image IMAGENAME=simple-game-server VERSION=<current-image-version>`
    - [x] `make bump-image IMAGENAME=simple-genai-game-server VERSION=<current-image-version>`
    - [x] `make bump-image IMAGENAME=supertuxkart-example VERSION=<current-image-version>`
    - [x] `make bump-image IMAGENAME=xonotic-example VERSION=<current-image-version>`

- [x] Create a PR for the above changes and send for review

- [x] After the above PR is approved, **before** merging it, run the following to generate and push the new example images:
    - [x] In `examples/allocation-endpoint`, run: `make cloud-build`
    - [x] In `examples/autoscaler-webhook`, run: `make cloud-build`
    - [x] In `examples/crd-client`, run: `make cloud-build`
    - [x] In `examples/custom-controller`, run: `make cloud-build`
    - [x] In `examples/simple-game-server`, run: `make cloud-build`
    - [x] In `examples/simple-genai-server`, run: `make cloud-build`
    - [x] In `examples/supertuxkart`, run: `make cloud-build`
    - [x] In `examples/xonotic`, run: `make cloud-build`

- [ ] Merge the above PR

### PR DESCRIPTION
**Pull request by 0xaravindh**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

 /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**: #3949

Closes #3949 

**Special notes for your reviewer**:

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module agones.dev/agones
 
-go 1.21
+go 1.22.0
```
Comment (peterzhongyi) [inline go.mod:3]: Does this need to be 1.22.6?

**Comment by 0xaravindh** [comment]
@gongmax  I’ve updated the Golang version as per the template. The PR is now ready for your review! #3988

**Review by gongmax — APPROVED** [review]
LGTM. Some minor comments for my learning purpose.

File: cmd/sdk-server/main.go [hunk 2]
```diff
@@ -245,6 +245,7 @@ func runGrpc(grpcServer *grpc.Server, grpcEndpoint string) {
 
 // runGateway runs the grpc-gateway
 func runGateway(ctx context.Context, grpcEndpoint string, mux *gwruntime.ServeMux, httpServer *http.Server) {
+	// nolint: staticcheck
```
[the remark this answers is not in the document]
Comment (0xaravindh) [reply cmd/sdk-server/main.go:248]: Upgrading the K8S package also required me to upgrade the gRPC package. So, the Dial and DialContext functions are now deprecated in version v1.65.0 (https://pkg.go.dev/google.golang.org/grpc#DialContext).

https://github.com/googleforgames/agones/blob/9e840c8238705750b9fef7ec0d7796c6029f0508/sdks/go/sdk.go#L75

File: cmd/allocator/main.go [hunk 3]
```diff
@@ -393,10 +393,10 @@ func runHTTP(listenCtx context.Context, workerCtx context.Context, h *serviceHan
 		if err == http.ErrServerClosed {
 			logger.WithError(err).Info("HTTP/HTTPS server closed")
 			os.Exit(0)
-		} else {
-			logger.WithError(err).Fatal("Unable to start HTTP/HTTPS listener")
-			os.Exit(1)
 		}
```
[the remark this answers is not in the document]
Comment (0xaravindh) [reply cmd/allocator/main.go:396]: Yes, this is required for lint upgrades.

File: sdks/swagger/beta.swagger.json [hunk 4]
```diff
@@ -260,6 +244,28 @@
     }
   },
   "definitions": {
+    "SDKAddListValueBody": {
```
[the remark this answers is not in the document]
Comment (0xaravindh) [reply sdks/swagger/beta.swagger.json:247]: The issue isn't related to linting. I believe this code was autogenerated (protoc-gen-go-grpc, protoc). After upgrading the gRPC package, changes were made to the generated code.

File: cmd/sdk-server/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (gongmax) [reply cmd/sdk-server/main.go:248]: We will see if we can fix it in a following PR

**Pull request merged**